commands/operator-sdk/cmd/build.go

```diff
@@ -42,12 +54,72 @@ For example:
 `,
 		Run: buildFunc,
 	}
+	buildCmd.Flags().BoolVarP(&enableTests, "enable-tests", "e", false, "Enable in-cluster testing by adding test binary to the image")
+	buildCmd.Flags().StringVarP(&testLocationBuild, "test-location", "t", "./test/e2e", "Location of tests")
+	buildCmd.Flags().StringVarP(&namespacedManBuild, "namespaced", "n", "deploy/operator.yaml", "Path of namespaced resources for tests")
+	return buildCmd
+}
+
+func verifyDeploymentImage(yamlFile []byte, imageName string) string {
```

Comment:
The `verifyDeploymentImage` function is a simple check to make sure that the the operator specified in the namespaced manifest file has the updated image name that is being used for the image currently being built. As mentioned in another comment, the namespaced manifest file may contain other resources, so we cannot decode the entire yaml as a deployment (although we may be able to do parts of it). There may potentially be multiple deployments in this file (the vault operator also has etcd-operator deployments in its namespaced manifest) and we shouldn't assume they all are identical (the possibility of other deployments is also the reason that this is just a warning; we don't want to fail if a user needs other deployments in addition to their operator). This is why we have the `ok` check for each level of the map; we don't want to panic if we encounter a deployment that doesn't quite match what our operator's deployment spec should look like (and even if we decoded to a `v1.Deployment` object, we would still need to have a `nil` check for each level, otherwise we might encounter a panic).

I'll add a comment above the function to describe its function a bit, since the name of the function isn't too clear.

Reply:
Think besides adding the description it would be useful to add a unit test for this as well.